include/chaiscript/dispatchkit/function_signature.hpp

```diff
@@ -4,121 +4,142 @@
 #include <type_traits>
 
 namespace chaiscript::dispatch::detail {
-  template<typename ... Param>
-    struct Function_Params
-    {
-    };
 
-  template<typename Ret, typename Params, bool IsNoExcept = false, bool IsMember = false, bool IsMemberObject = false, bool IsObject = false>
-    struct Function_Signature {
-      using Param_Types = Params;
-      using Return_Type = Ret;
-      constexpr static const bool is_object = IsObject;
-      constexpr static const bool is_member_object = IsMemberObject;
-      constexpr static const bool is_noexcept = IsNoExcept;
-      template<typename T>
-        constexpr Function_Signature(T &&) noexcept {}
-      constexpr Function_Signature() noexcept = default;
-    };
+template<typename... Param>
+struct Function_Params
```

Comment:
With VS2019, I get a compilation error because of ambiguity between `chaiscript::dispatch::detail::Function_Params` here and the "regular" `chaiscript::Function_Params`.

I "solved" it locally by renaming `chaiscript::dispatch::detail::Function_Params` to be `Function_Params_t` (or anything different).  Alternatively, in a few places you could solve it by referring explicitly to `chaiscript::Function_Params` (instead of just `Function_Params` to disambiguate between these two types (e.g. in the "detail" helper functions in `register_function.hpp`). Only then I was able to get things compiling on VS2019.